src/rgw/driver/rados/rgw_data_sync.cc

```diff
@@ -4654,6 +4669,16 @@ int RGWBucketFullSyncCR::operate(const DoutPrefixProvider *dpp)
           }
           return set_cr_error(-ECANCELED);
         }
+        // for testing purpose to slow down the execution pace of the this loop
+        if (cct->_conf->rgw_inject_delay_sec > 0 &&
+            std::string(cct->_conf->rgw_inject_delay_pattern) == "delay_bucket_full_sync_loop") {
```

Comment:
nit: this is is creating a temporary string on every iteration even when delay is disabled.
```suggestion
        if (cct->_conf->rgw_inject_delay_sec > 0) {
            if (std::string_view(cct->_conf->rgw_inject_delay_pattern)
            == "delay_bucket_full_sync_loop") {
```